modules/hostap/src/supp_api.c

```diff
@@ -905,6 +910,27 @@ int supplicant_channel(const struct device *dev, struct wifi_channel_info *chann
 	return wifi_mgmt_api->channel(dev, channel);
 }
 
+int supplicant_get_wifi_conn_params(const struct device *dev,
+			struct wifi_connect_req_params *params)
+{
+	struct wpa_supplicant *wpa_s;
+	int ret = 0;
+
+	k_mutex_lock(&wpa_supplicant_mutex, K_FOREVER);
+
+	wpa_s = get_wpa_s_handle(dev);
+	if (!wpa_s) {
+		ret = -1;
+		wpa_printf(MSG_ERROR, "Device %s not found", dev->name);
+		goto out;
+	}
+
+	params = &last_wifi_conn_params;
```

Comment:
this would still be a shallow-copy, so, may be use memcpy

Reply:
updated